Tests/Unit/MSFT_xIISFeatureDelegation.tests.ps1

```diff
@@ -4,14 +4,49 @@ $global:DSCResourceName = 'MSFT_xIISFeatureDelegation'
 #region HEADER
 
 [String] $moduleRoot = Split-Path -Parent (Split-Path -Parent (Split-Path -Parent $Script:MyInvocation.MyCommand.Path))
- if ( (-not (Test-Path -Path (Join-Path -Path $moduleRoot -ChildPath 'DSCResource.Tests'))) -or `
-      (-not (Test-Path -Path (Join-Path -Path $moduleRoot -ChildPath 'DSCResource.Tests\TestHelper.psm1'))) )
+$repoSource = (Get-Module -Name $global:DSCModuleName -ListAvailable)
+
+if (($repoSource -ne $null) -and ($repoSource[0].RepositorySourceLocation.Host -eq 'www.powershellgallery.com'))
+{
+    if ( -not (Test-Path -Path (Join-Path -Path $moduleRoot -ChildPath 'Tests\DscResourceTestHelper')) )
+    {
+
+        $choice = read-host "In order to run this test you need to install a helper module, continue with installation? (Y/N)"
+
+        if ($choice -eq "y"){
+            #install test folders from gallery
+            Save-Module -Name 'DscResourceTestHelper' -Path (Join-Path -Path $moduleRoot -ChildPath 'Tests')
+        }
+    
+        else 
+        {
+            Write-Host "Unable to run tests without the required helper module - Exiting test"
+            return
+        }
+        
+    }
+
+    $testModule = Get-Module -Name 'DscResourceTestHelper' -ListAvailable
```

Comment:
Have you tested this path? I believe this won't work since module isn't in PSModulePath. You can use Find-Module before Save-Module to get version